addons/web/static/tests/core/file_input_tests.js

```diff
@@ -138,4 +139,30 @@ QUnit.module("Components", ({ beforeEach }) => {
         await editInput(target, ".o_file_input input", file);
         assert.verifySteps(["fake_file.txt"], "file has been uploaded a second time");
     });
+
+    QUnit.test("Upload button is disabled if attachment upload is not finished", async (assert) => {
+        assert.expect(2);
+
+        const uploadedPromise = makeDeferred();
+        await createFileInput({
+            mockPost: async (route, params) => {
+                if (route === "/web/binary/upload_attachment") {
+                    await uploadedPromise;
+                }
+                return "[]";
+            },
+            props: {},
+        });
+        //enable button
+        const input = target.querySelector(".o_file_input input");
+        await triggerEvent(input, null, "change", {}, { skipVisibilityCheck: true });
+
+        //disable button
+        assert.ok(input.disabled,
+            "the upload button should be disabled on upload");
+
+        await afterNextRender(() => uploadedPromise.resolve());
```

Comment:
the web module doesn't depend on mail, you can't use this test util. You can just resolve the promise and `await nextTick()` instead.

Reply:
Hello @sdegueldre 
Changes are done!